client/tailscale/tailscale.go

```diff
@@ -93,25 +90,11 @@ func (c *Client) setAuth(r *http.Request) {
 	}
 }
 
-// nodeKeyAuth is an AuthMethod for NewClient that authenticates requests
-// using a node key over the Noise protocol.
-type nodeKeyAuth key.NodePublic
-
-func (k nodeKeyAuth) modifyRequest(req *http.Request) {
-	// QueryEscape the node key since it has a colon in it.
-	nk := url.QueryEscape(key.NodePublic(k).String())
-	req.SetBasicAuth(nk, "")
-}
-
-// NewNoiseClient is a convenience method for instantiating a new Client
-// that uses the Noise protocol for authentication.
-//
-// tailnet is the globally unique identifier for a Tailscale network, such
-// as "example.com" or "user@gmail.com".
-func NewNoiseClient(tailnet string, noiseRoundTripper http.RoundTripper, nk key.NodePublic) *Client {
+// NewTSNetNoiseClient returns a  new Client that uses the Noise protocol for
+// authentication for use within a tsnet binary.
+func NewTSNetNoiseClient(noiseRoundTripper http.RoundTripper) *Client {
```

Comment:
why is this TSNet specific?

Reply:
Because it requires you to be running in-process in tailscaled to be able to get a suitable RoundTripper from LocalBackend.

This doesn't yet work from another process (like cmd/tailscale CLI) because we never proxy noise stuff through tailscaled where it can grab a stateful noise conn to sign.

I ripped out all that "modify the *http.Request as an auth method" stuff which doesn't support stateful connections (you can't sign a request to a specific connection if you don't have the connection yet)

Reply:
oh right, I forgot 🤦🏼‍♂️

Reply:
maybe call it something else though? so that we don't have to rename later.